### PR TITLE
Update core translations and create new set site_kb for knowledge base

### PR DESCRIPTION
**Pull request by alex-mtl**
update for rich editor and some other parts of front end translations needed (scan library/src/scripts)

Relates to: https://github.com/vanilla/vanilla/issues/8630

### CODE REVIEW

File: tx-source/site_core.php [hunk 1]
```diff
@@ -1775,3 +1775,84 @@
...
+$Definition["PPT"] = "PPT";
+$Definition["Zip"] = "Zip";
+$Definition["Expand All"] = "Expand All";
+$Definition["Collapse All"] = "Collapse All";
+$Definition["Closed Folder"] = "Closed Folder";
+$Definition["Open Folder"] = "Open Folder";
+$Definition["Article"] = "Article";
+$Definition["Me"] = "Me";
+$Definition["New Folder"] = "New Folder";
+$Definition["Ok"] = "Ok";
```
Comment (tburry) [inline tx-source/site_core.php]: This file already has 'OK'.

File: tx-source/site_core.php [hunk 2]
```diff
@@ -1775,3 +1775,84 @@
...
+$Definition["Menu"] = "Menu";
+$Definition["My Account"] = "My Account";
+$Definition["You do not have any messages yet."] = "You do not have any messages yet.";
+$Definition["Drafts"] = "Drafts";
+$Definition["Mark All Read"] = "Mark All Read";
+$Definition["Messages: "] = "Messages: ";
+$Definition["Article Revisions"] = "Article Revisions";
+$Definition["My Account Tab"] = "My Account Tab";
+$Definition["Notifications: "] = "Notifications: ";
+$Definition["Sign in"] = "Sign in";
```
Comment (tburry) [inline tx-source/site_core.php]: This file already has "Sign In".

File: tx-source/site_core.php [hunk 3]
```diff
@@ -1775,3 +1775,84 @@
...
+$Definition["Last Updated: <0/>"] = "Last Updated: <0/>";
+$Definition["No results for '<0/>'."] = "No results for '<0/>'.";
+$Definition["Published <0/> by <1/>"] = "Published <0/> by <1/>";
+$Definition["Updated <0/> by <1/>"] = "Updated <0/> by <1/>";
+$Definition["Search for <0/>"] = "Search for <0/>";
+$Definition["<0/> message"] = "<0/> message";
+$Definition["<0/> messages"] = "<0/> messages";
+$Definition["Uploaded <0/>"] = "Uploaded <0/>";
+$Definition["<0/> by <1/>"] = "<0/> by <1/>";
+$Definition["Search Articles"] = "Search Articles";
```
Comment (tburry) [inline tx-source/site_core.php]: Wouldn't these belong in site_kb?

File: tx-source/site_core.php [hunk 4]
```diff
@@ -241,11 +248,14 @@
 $Definition['Are you sure you want to leave this conversation?'] = 'Are you sure you want to leave this conversation?';
 $Definition['Are you sure you want to leave this group?'] = 'Are you sure you want to leave this group?';
 $Definition['Are you sure you wish to continue?'] = 'Are you sure you wish to continue?';
+$Definition["Article"] = "Article";
```
Comment (charrondev) [inline tx-source/site_core.php]: These source strings are being moved into knowledge base.

https://github.com/vanilla/vanilla/pull/8756
https://github.com/vanilla/knowledge/pull/967

We should move them into the knowledge resource.

File: tx-source/site_kb.php [hunk 5]
```diff
@@ -0,0 +1,136 @@
+<?php
+
+$Definition[" by "] = " by ";
```
Comment (charrondev) [inline tx-source/site_kb.php]: I've deleted the file that contains this translation. We should remove it. https://github.com/vanilla/knowledge/pull/967

**Pull request merged**